language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -237,14 +249,63 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
         // That is, on the first traversal, we create opaque structs (i.e., partially formed,
         // deferring field translation). The second traversal will then fill in the struct bodies
         // where it will have all structure types previously defined.
-        for s_env in &all_structs {
+        for (s_env, _) in &all_structs {
```

Comment:
I meant to ask this before but it is even better time now.
Can we have "nested" generics?
Then probably in one pass we may still have unsubstituted opaques and should keep iterating?

Reply:
If you mean like the following, yes:
 ``` 
    struct Foo<T> has copy, drop { x: T }
    struct Baz<T1, T2> has copy, drop {
        x: T1,
        y: Foo<T2>,
    }
```
Or if you mean like the following, yes also:
```
fun tryit(): () {
        let t1 = Bar<u8, Foo<u64>> { x: 123, y: Foo<u64> { x: 456 } };
    }
```
Being an "initial" support patch, I can't claim every possible scenario will work, and obviously the testing is minimal. Once we start putting more code through (e.g., the standard library) and bigger things from the Aptos tree, we'll address any limitations. Just writing the test cases takes me almost as long as writing the patch itself.

But this is an excellent point-- iteration vs laziness. After having written the code this way, a lazy/deferred approach with recursion may be a more natural (possibly simpler) solution. The goal today is to get at least basic generic support going and handle the common cases.

Reply:
I agree, let's push it and then extend the coverage, probably case by case.
It will be easier to tests the cases if we have some core working.